### PR TITLE
add a layout web component for css properties

### PR DESCRIPTION
**Pull request by janechu**
# Pull Request

## 📖 Description

This creates a CSS layout web component in `@microsoft/fast-tooling` that can be used by itself or with the experimental custom css control in `@microsoft/fast-tooling-react`. It primarily uses the CSS flexbox to update CSS either by the change event or an `onChange` property assigned to the element.

This PR also contains the integration into the Creator design tool.

Screen Shot 2021-05-28 at 5 49 02 PM

## 👩‍💻 Reviewer Notes

Running the Creator site will show off the control in the right pane, to view it by itself run `yarn start` to use the manual testing app in `@microsoft/fast-tooling` package.

## 📑 Test Plan

New unit tests have been added for this component.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
Could you post a screenshot? 😁

**Comment by EisenbergEffect** [comment]
Sweet!

**Review by EisenbergEffect — APPROVED** [review]
As I was reviewing, I realized...I can't remember reviewing another web components you've written. Is this your first FAST Web Component @janechu ?

**Comment by EisenbergEffect** [comment]
We should get this in, but there may be some way to reuse some FAST components to improve this a bit in the future. In particular, I'm thinking of replicating the pattern for how the color swatch was built for the FAST site. That's kind of similar to the radio group-like situation with the icon buttons here. We shouldn't block on that though. We can revisit when there's more time.

File: packages/tooling/fast-tooling/src/web-components/css-layout/css-layout.template.ts [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+    </svg>
+`;
+
+/**
+ * The template for the css-layout component.
+ * @public
+ */
+export const CSSLayoutTemplate = html<CSSLayout>`
+    <template>
+        <div class="root">
```
Comment (chrisdholt) [inline packages/tooling/fast-tooling/src/web-components/css-layout/css-layout.template.ts:538]: I could be missing an underlying need for this, but from what I can tell, you technically could remove this and the ‘template’ and use ‘host:’ where your root selector is in CSS

**Review by chrisdholt — APPROVED** [review]

**Comment by janechu** [comment]
> As I was reviewing, I realized...I can't remember reviewing another web components you've written. Is this your first FAST Web Component @janechu ?

It very well might be, I've reviewed and looked at web components but I also can't remember actually writing one before this.

File: packages/tooling/fast-tooling/src/web-components/css-layout/css-layout.template.ts (hunk 1 above)
Comment (janechu) [reply packages/tooling/fast-tooling/src/web-components/css-layout/css-layout.template.ts:538]: Nope, this was some leftover bits while I was revving that now no longer serve a purpose, I'll remove.

**Pull request merged**